File: src/main/java/com/example/librarymanagement/repository/BookRepository.java
package com.example.librarymanagement.repository;

import com.example.librarymanagement.model.entity.Book;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface BookRepository extends JpaRepository<Book, Long> {
    Boolean existsByIsbn(Long isbn);
    Optional<Book> findBookByTitleAndAuthorId(String title, Long authorId);
}

File: src/main/java/com/example/librarymanagement/repository/BookRepository.kt
package com.example.librarymanagement.repository

import com.example.librarymanagement.model.domain.Book
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface BookRepository{
    fun save(book: Book): Mono<Book>
    fun findById(bookId: String): Mono<Book>
    fun findAll(): Flux<Book>
    fun deleteById(bookId: String): Mono<Unit>
    fun existsById(bookId: String): Mono<Boolean>
    fun existsByIsbn(isbn: Long): Mono<Boolean>
    fun findBookByTitleAndAuthorId(title: String, authorId: String): Flux<Book>
}
